common/src/main/java/org/ergoplatform/appkit/impl/ErgoTreeContract.java

```diff
@@ -30,4 +33,8 @@ public ErgoTreeContract substConstant(String name, Object value) {
     public Values.ErgoTree getErgoTree() {
         return _ergoTree;
     }
+
+    @Override
+    public Address getAddress(NetworkType networkType) { return Address.fromErgoTree(_ergoTree, networkType); }
```

Comment:
Yeah, networkType is used by the compiler to decode PK("address") expressions. See https://github.com/ScorexFoundation/sigmastate-interpreter/blob/a3068e8b7c497a6dd1b7871708a7473c5fe0c830/sigmastate/src/main/scala/sigmastate/lang/SigmaPredef.scala#L153
And networkType is not part of ErgoTree, so this asymmetry between ErgoContract implementations because of the compiler.
I would suggest to remove this methods, as they don't encapsulate much, but create confusion and ambiguity.
But as you mentioned, it would be useful to have direct method `ErgoContract.getAddress` because yes,  contracts and addresses are kind of the same thing.
However, this method will require passing networkPrefix to the ErgoTreeContract constructor, and I think that is ok and it can be added.

Reply:
Yeah, passing in NetworkType into ErgoTreeContract seems like the best way to go about this. I'll work on implementing this today.